reaper/cluster.go

```diff
@@ -0,0 +1,243 @@
+	Cluster *Cluster
+	Error   error
+}
+
+type RepairSchedule struct {
+	Id                string    `json:"id"`
+	Owner             string    `json:"owner,omitempty"`
+	State             string    `json:"state,omitempty"`
+	Intensity         float64   `json:"intensity,omitempty"`
+	ClusterName       string    `json:"cluster_name,omitempty"`
+	KeyspaceName      string    `json:"keyspace_name,omitempty"`
+	RepairParallism   string    `json:"repair_parallelism,omitempty"`
+	IncrementalRepair bool      `json:"incremental_repair,omitempty"`
+	ThreadCount       int       `json:"repair_thread_count,omitempty"`
+	UnitId            string    `json:"repair_unit_id,omitempty"`
+	DaysBetween       int       `json:"scheduled_days_between,omitempty"`
+	Created           time.Time `json:"creation_time,omitempty"`
+	Paused            time.Time `json:"pause_time,omitempty"`
+	NextActivation    time.Time `json:"next_activation,omitempty"`
+}
+
+// All the following types are used internally by the client and not part of the public API
+
+type clusterStatus struct {
+	Name           string     `json:"name"`
+	JmxUsername    string     `json:"jmx_username,omitempty"`
+	JmxPasswordSet bool       `json:"jmx_password_is_set,omitempty"`
+	Seeds          []string   `json:"seed_hosts,omitempty"`
+	NodeStatus     nodeStatus `json:"nodes_status"`
+}
+
+type nodeStatus struct {
+	EndpointStates []gossipStatus `json:"endpointStates,omitempty"`
+}
+
+type gossipStatus struct {
+	SourceNode    string   `json:"sourceNode"`
+	EndpointNames []string `json:"endpointNames,omitempty"`
+	TotalLoad     float64  `json:"totalLoad,omitempty"`
+	Endpoints     map[string]map[string][]endpointStatus
+}
+
+type endpointStatus struct {
+	Endpoint       string  `json:"endpoint"`
+	DataCenter     string  `json:"dc"`
+	Rack           string  `json:"rack"`
+	HostId         string  `json:"hostId"`
+	Status         string  `json:"status"`
+	Severity       float64 `json:"severity"`
+	ReleaseVersion string  `json:"releaseVersion"`
+	Tokens         string  `json:"tokens"`
+	Load           float64 `json:"load"`
+}
+
+func (c *client) GetClusterNames(ctx context.Context) ([]string, error) {
+	res, err := c.doGet(ctx, "/cluster", nil, http.StatusOK)
+	if err == nil {
+		clusterNames := make([]string, 0)
+		err = c.readBodyAsJson(res, &clusterNames)
+		if err == nil {
+			return clusterNames, nil
+		}
+	}
+	return nil, fmt.Errorf("failed to get cluster names: %w", err)
+}
+
+func (c *client) GetCluster(ctx context.Context, name string) (*Cluster, error) {
+	path := "/cluster/" + url.PathEscape(name)
+	res, err := c.doGet(ctx, path, nil, http.StatusOK)
+	if err == nil {
+		clusterStatus := &clusterStatus{}
+		err = c.readBodyAsJson(res, clusterStatus)
+		if err == nil {
+			return newCluster(clusterStatus), nil
+		}
+	}
+	return nil, fmt.Errorf("failed to get cluster %s: %w", name, err)
+}
+
+// Fetches all clusters. This function is async and may return before any or all results are
```

Comment:
One Go commenting standard I learned, not sure if it's essential for us, is to start the comment with the name of the function.  

// GetClusters fetches all clusters ....

Reply:
Correct. I did not want to change the Cluster methods though in this PR, I simply copied them from the old file over to the new one.

Reply:
OK I changed the comments.